src/Components/MultiInput/Component.purs

```diff
@@ -215,6 +232,28 @@ handleOnBlur ::
   ComponentM m Unit
 handleOnBlur index = do
   commitEditing index
+  Halogen.raise (On Blur)
+
+handleOnFocus ::
+  forall m.
+  MonadAff m =>
+  Int ->
+  ComponentM m Unit
+handleOnFocus index = do
+  old <- Halogen.get
+  void $ Control.Monad.Maybe.Trans.runMaybeT do
+    item <-
+      Control.Monad.Maybe.Trans.MaybeT <<< pure $ do
+        Data.Array.index old.items index
+    Control.Monad.Maybe.Trans.lift do
+      case item of
+        Display _ -> pure unit
+        Edit { inputBox: { text } } -> do
+          Halogen.raise (On (ValueInput text ))
+          Halogen.raise (On Focus)
+        New { inputBox: { text }} -> do
+          Halogen.raise (On (ValueInput text ))
+          Halogen.raise (On Focus)
```

Comment:
🎨  This seems like it could be simplified with:
```purs
case Data.Array.index old.items index of
  Nothing -> pure unit
  Just item -> case item of
    Display _ -> pure unit
    Edit { inputBox: { text } } -> do
      Halogen.raise (On (ValueInput text))
      Halogen.raise (On Focus)
    New { inputBox: { text }} -> do
      Halogen.raise (On (ValueInput text))
      Halogen.raise (On Focus)
```